### ISSUE TITLE
album covers not displayed correctly

### ISSUE THREAD

**Issue by joselao** [music-assistant/hass-music-assistant#1253]
### What version of Music Assistant has the issue?

2.0.0b30

### The problem

the covers of local files are not correctly visualised, in addition the covers show in the browser are different that showed in iphone app.
see example images below (PC browser & app), the covers are the same.

![Captura de pantalla 2023-05-18 a las 10 22 04](https://github.com/music-assistant/hass-music-assistant/assets/8224312/2c111918-3b93-4f25-b7ac-91d33a3143ae)

![IMG_4043](https://github.com/music-assistant/hass-music-assistant/assets/8224312/a2988976-e6d9-47f2-acf4-7976fd1bb8d4)

### How to reproduce

add music files to media and sync MA. the go to discs and covers are not all show. The covers are inside each file in the metadata:

![Captura de pantalla 2023-05-18 a las 12 03 09](https://github.com/music-assistant/hass-music-assistant/assets/8224312/50e4cad0-af65-49bb-9e6e-2cf4c6f2f7e1)

### What version of Home Assistant Core are your running

2023.4.6

### What type of installation are you running?

Home Assistant Supervised

### On what type of hardware are you running?

Linux

**Comment by OzGav** [comment]
How are you embedding the cover art?

**Comment by OzGav** [comment]
Actually given that different browsers are behaving differently hen quite possibly a caching issue. See the GitHub FAQs if you don’t know how to clear the browser cache

**Comment by joselao** [comment]
> How are you embedding the cover art?

with iTunes (apple).

the Jellyfin reads correctly the same media folder:
![IMG_4044](https://github.com/music-assistant/hass-music-assistant/assets/8224312/6009e13c-182a-4cbc-8dad-0aed8029bda4)

**Comment by OzGav** [comment]
Have you tried clearing the browser cache?

**Comment by OzGav** [comment]
Additionally MA loads a lot on demand and slowly in the background so try visiting an album and clicking on the refresh button

**Comment by OzGav** [comment]
Did you try the “Refresh Item” option that is found by clicking on the three dots in the top right?

**Comment by joselao** [comment]
yes!! I did "refresh item" inside an specific album and also "actualise list of elements": 
![2](https://github.com/music-assistant/hass-music-assistant/assets/8224312/5692841f-d9ca-43e6-89f2-5488d3ac8347)


and "synchronize now" in disc section: 
![1](https://github.com/music-assistant/hass-music-assistant/assets/8224312/d7c1bd01-d0ca-4298-8b7c-971073c494ad)

nothing happens...

**Comment by OzGav** [comment]
Then maybe there is an issue with the way iTunes is embedding. (I know jellyfin works but…) try tagging and embedding an image in one of your albums using MusicBrainz Picard.

**Comment by OzGav** [comment]
But again please check the logs for error messages related to this.

**Comment by erkr** [comment]
I guess it's a frontend issue. The fact some browsers display the images, and others don't, suggests that the art info is recognised by MA, but the rendered browser page is not consistently displayed on the various output

**Comment by marcelveldt** [comment]
I'm having a hard time understanding this issue. Am I reading it correctly that the covers are working on a mobile browser but not on a desktop browser ?

**Comment by joselao** [comment]
I Marcel,

The problem seems that covers embedded with iTunes are not fully displayed in Music Assistant (2.0.0b30) using:

- opening HA in local IP and using a browser (Chrome) on a computer
- nor the mobile (iPhone) using the HA app

To check if the cover images are correctly embedded, I tested the "Jellifyn Sever addon" reading the same media folder to verify if the covers are fully displayed using Jellifyn, and yes they are fully displayed using the HA app on the mobile phone and also the PC browser.

I have not checked yet to embed covers using "MusicBrainz Picard" as OzGav suggested.

regards
Jose

**Comment by erkr** [comment]
@joselao now you are comparing with the Jellifyn server add on, but if I understand correctly, you mentioned earlier that there was a difference in missing covers in the chrome app vs the companion (mobile) app. 
Is that the case or not?!

**Comment by joselao** [comment]
@erkr yes, that's it, I got different covers in the chrome app vs the companion (mobile) app, that's the first and second screen shots at the beginning of the issue.

**Comment by OzGav** [comment]
Any update to this with b35. There are no other reports of this so far...?

**Comment by joselao** [comment]
Hi,

yes I'm reporting my last steps:

1) updated to: Home Assistant 2023.6.1 (supervisor)
2) updated music assistant to: 2.0.0b35
3) ordering the 475 albums alphabetically (screenshot from PC browser chrome), no covers here:
![Captura de pantalla 2023-06-12 a las 11 22 09](https://github.com/music-assistant/hass-music-assistant/assets/8224312/2a716b11-a687-4571-8f61-3eb207bb13b3)

4) deleting from the media folder all the albums from B to Z, only leaving A letter (23 albums) in order to check if the problem is the amount of albums/files or my limited RAM (4Gb):
5) restarting the MASS add-on and synchronizing + actualizing all the albums
6) MASS shows same 475 albums mainly without covers (must show 23 remember!)
7) cleaning the browser cache -> no results same as step 6)
8) disabling in MASS the "Filesystem (local disk)" in music providers, restarting the MASS addon + synchronizing + actualizing albums
9) going to albums and same as step 6) I show the 475 albums mainly without covers
-- at that point -- I think the MASS database doesn't reflect the content of the media folders
10) unistalling MASS addon and re-installing it (V 2.0.0b35), adding local folder provider, at this point all except one album have covers:
![Captura de pantalla 2023-06-12 a las 12 38 10](https://github.com/music-assistant/hass-music-assistant/assets/8224312/4c7cd054-a947-4320-859e-8060795f0858)
-- all the covers have been included using iTunes, not "MusicBrainz Picard" yet. --

using the "Companion app" now i'm getting the same result as PC browser.

11) so I will check the size of the image of this missing cover because this album has really a cover:
![Captura de pantalla 2023-06-12 a las 12 41 16](https://github.com/music-assistant/hass-music-assistant/assets/8224312/0f0ffaf7-5693-4a4a-ae0d-66b338b26274)

I open iTunes and extract the picture of the covers in order to check the image size:

- dimensions: 240 × 240
- color space: RGB
- class: jpg
- size: 13.499 bytes (16 KB in disk)

seems OK (not too big file) from my point of view

12) I installed "MusicBrainz Picard", and actualized all the covers of the first 23 albums, and then I click "actualize" the albums in MASS and I got finally the last missing cover here!:

![Captura de pantalla 2023-06-12 a las 13 06 30](https://github.com/music-assistant/hass-music-assistant/assets/8224312/42152e49-1b29-443e-9f5d-533a7d550fbd)

13) At this point the conclusions are:
- MASS prefers the covers included using "MusicBrainz Picard" than others like Apple iTunes as OzGav realised. I don't know if this a issue related only to a limited amount of RAM (>= 4Gb) or the way the covers are embebed
- MASS doesn't actualize the DDBB when files have been deleted, it can be a new issue?, I needed to unsistall Music Assistant in order to refresh the data base (step 10)

14) Finally I will try to check if the covers of my full 475 albums can be managed by MASS if they have been embebed using  
 "MusicBrainz Picard" instead of iTunes.... the results... soon....

**Comment by OzGav** [comment]
MA loads the covers slowly to avoid bogging down your system. So before you do too much work you should be able to do the following:
1. Add the music provider
2. Observe that no covers are shown
3. Click on an album
4. Wait for the icon that looks like an arrow bent into a circle (refresh icon) to have a red dot appear beside it. Click that icon
5. Cover art should now show on the album screen if it wasn't shown before.
6. Return to the album screen and that album should have the cover thumbnail
7. Do that for all your albums OR just wait! MA should fetch one cover about every 30 seconds.

If the covers never show after waiting and you cannot load them by using the  refresh icon then indeed it suggests to me that there is something wrong with the embedded artwork. You can also just put the artwork in the album folder, see here https://github.com/orgs/music-assistant/discussions/820 for how to best do that. Personally I generally don't embed the artwork because that is just increasing the file size of all your tracks when you could just have one image file in the folder

**Comment by OzGav** [comment]
Also if the red dot never appears try using the three dot menu in the top right and select REFRESH ITEM

**Comment by OzGav** [comment]
Thanks for the great explanation of what you are doing by the way it really helps understand the issue!

**Comment by joselao** [comment]
> OzGav

thanks to you and all developers of MA + HA for this wonderful ecosystem!!

**Comment by joselao** [comment]
continuing the issue:

14) I introduced all the albums from "B" to "Z", surprisingly the majority (about 90%/95%) of the covers (embedded with iTunes) have been displayed correctly!!!:

![Captura de pantalla 2023-06-13 a las 8 54 27](https://github.com/music-assistant/hass-music-assistant/assets/8224312/0b4bcc84-bd87-465a-b104-459e7525bd8a)

only few albums doesn't display any cover.

I tried to do as OzGav commented before: click on the "refresh icon" in each album without cover but no results. so I will re-introduce the covers for these few missing cover albums with "MusicBrainz Picard".

Conclusion:
- it doesn't seems a RAM issue (4Gb in my case)
- I think it was a previous version of MA (2.0.0b30) that filled the covers of the DDBB in a different way that the last one version (2.0.0b35), or the new version has improvements that performs better.

Thanks for your help!!

**Comment by OzGav** [comment]
OK if this is working to your satisfaction I will close this one.

**Comment by OzGav** [comment]
Before you go though did you get any errors in the log when you tried to refresh the item?

**Comment by joselao** [comment]
hope it helps:

`          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/PIL/Image.py", line 3298, in open
    raise UnidentifiedImageError(msg)
PIL.UnidentifiedImageError: cannot identify image file <_io.BytesIO object at 0x7f783444cef0>
2023-06-13 09:32:30 ERROR aiohttp.server  -- Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/webserver.py", line 115, in _handle_catch_all
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/metadata.py", line 338, in _handle_imageproxy
    image_data = await self.get_thumbnail(path, size=size, provider=provider)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/metadata.py", line 322, in get_thumbnail
    thumbnail = await get_image_thumb(self.mass, path, size=size, provider=provider)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/helpers/images.py", line 49, in get_image_thumb
    return await asyncio.to_thread(_create_image)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/helpers/images.py", line 43, in _create_image
    img = Image.open(BytesIO(img_data))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/PIL/Image.py", line 3298, in open
    raise UnidentifiedImageError(msg)
PIL.UnidentifiedImageError: cannot identify image file <_io.BytesIO object at 0x7f7836b45940>
2023-06-13 09:32:30 ERROR aiohttp.server  -- Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/webserver.py", line 115, in _handle_catch_all
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/metadata.py", line 338, in _handle_imageproxy
    image_data = await self.get_thumbnail(path, size=size, provider=provider)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/metadata.py", line 322, in get_thumbnail
    thumbnail = await get_image_thumb(self.mass, path, size=size, provider=provider)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/helpers/images.py", line 49, in get_image_thumb
    return await asyncio.to_thread(_create_image)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/helpers/images.py", line 43, in _create_image
    img = Image.open(BytesIO(img_data))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/PIL/Image.py", line 3298, in open
    raise UnidentifiedImageError(msg)
PIL.UnidentifiedImageError: cannot identify image file <_io.BytesIO object at 0x7f7836b45940>
2023-06-13 09:32:30 ERROR aiohttp.server  -- Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/webserver.py", line 115, in _handle_catch_all
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/metadata.py", line 338, in _handle_imageproxy
    image_data = await self.get_thumbnail(path, size=size, provider=provider)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/metadata.py", line 322, in get_thumbnail
    thumbnail = await get_image_thumb(self.mass, path, size=size, provider=provider)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/helpers/images.py", line 49, in get_image_thumb
    return await asyncio.to_thread(_create_image)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/helpers/images.py", line 43, in _create_image
    img = Image.open(BytesIO(img_data))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/PIL/Image.py", line 3298, in open
    raise UnidentifiedImageError(msg)
PIL.UnidentifiedImageError: cannot identify image file <_io.BytesIO object at 0x7f7836b45ad0>`

**Comment by OzGav** [comment]
Have you had a chance to re-embed the images in the tracks?

**Comment by joselao** [comment]
Hi,

yes!!, the problem was solved, Thanks!!

I think I found the issue, it seems that iTunes is embedding the images tagged as stickers, and also allow to embed more than one image. Whereas Picard seems only embed one image tagged has "Front", below screenshots from the re-embedding of Picard:

![Captura de pantalla 2023-06-21 a las 12 17 46](https://github.com/music-assistant/hass-music-assistant/assets/8224312/a7f477b0-ef31-4454-aff7-33204bc24206)

![Captura de pantalla 2023-06-21 a las 12 18 02](https://github.com/music-assistant/hass-music-assistant/assets/8224312/bdae2675-b933-498e-bbe5-272f686444c7)

![Captura de pantalla 2023-06-21 a las 12 21 03](https://github.com/music-assistant/hass-music-assistant/assets/8224312/d25e37c1-6da9-4419-92db-7e9c5769728f)

![Captura de pantalla 2023-06-21 a las 12 21 28](https://github.com/music-assistant/hass-music-assistant/assets/8224312/11811033-476b-4787-a1f0-eca3e930cb5c)

Maybe MA looks for "Front" tag and if there are images with that tag doesn't display any image?

In any case Picard solves the problem!!, but if you have a huge discography can be a big effort.

Thanks for your help!!

Jose
